### ISSUE TITLE
Unable to add device

### ISSUE THREAD

**Issue by Bayharbourbutcher1** [dmunozv04/iSponsorBlockTV#137]
Device - Apple tv 4k, TCL android tv
OS - Tvos 17.3 and Android tv

I have not been able to add device to the config since moving to mac from windows. I have all the requirements installed but whenever i enter the tv code from the youtube app into the gui it errors out 'failed to add device'

I have removed the install completely and downloaded fresh from releases but the issue persists. I have generated multiple codes from youtube apps but no success. Please assist. I had this setup working fine on my windows 11 prior to this

**Comment by dmunozv04** [comment]
Are you running the code from source?
 Maybe the python version you're using is somehow not compatible with iSponsorBlockTV. Would you mind sending the output of `python --version` or `python3 --version`?

**Comment by Bayharbourbutcher1** [comment]
> Are you running the code from source?
>  Maybe the python version you're using is somehow not compatible with iSponsorBlockTV. Would you mind sending the output of `python --version` or `python3 --version`?

Yeah i am running it from the source. My python version is 3.12.1

I also tried the new release today but sadly the issue persists.

**Comment by dmunozv04** [comment]
@matzZz are you using the tui config editor or the cli one? Are you trying to use auto discover or manually enter the device code?

**Comment by matzZz** [comment]
I used the tui config "add with lan discovery". but It shows nothing...
I had only two Apple TV.

**Comment by dmunozv04** [comment]
@matzZz for lan discovery to work on Apple TV you need to:
1) Force close the app and reopen it
2) Go to the lan pairing section in settings
3) Hope it works

I learnt (after creating the logic for autodescovery) that it is very unreliable and thus I'd suggest you add it manually with the pairing code

**Comment by matzZz** [comment]
now it works. tanks! :)

**Comment by dmunozv04** [comment]
@matzZz that's great to hear!

**Comment by benklinger** [comment]
Had the same issue. Trying re-installing iSponsorblock, tried upgrading python.
In the end what worked:
1. Open setup
2. Un-toggle what to filter
3. Save and quit
4. Re-open setup
5. Just select one filter
6. Add Device with TV code
7. Select all filters
